File: Swift-C++-IR-Comparison/test/functions/Swift/FunctionFunctionTypeAsReturn.swift
@main
struct MyMain {

    static func main() {
        var currentValue = 3
        let moveNearerToZero = chooseStepFunction(backward: currentValue > 0)
    }

}

func chooseStepFunction(backward: Bool) -> (Int) -> Int {
    return backward ? stepBackward : stepForward
}

func stepForward(_ input: Int) -> Int {
    return input + 1
}
func stepBackward(_ input: Int) -> Int {
    return input - 1
}

File: Swift-C++-Testsuite/test/functions/C++/FunctionFunctionTypeAsReturn.cpp

 #include<iostream>
 
 int stepForward(int input) {
     return input + 1;
 }

 int stepBackward(int input) {
     return input - 1;
 }
 
 std::function<int (int)> chooseStepFunction(bool backward) {
      return backward ? stepBackward: stepForward;
 }
 

 
 int main() {
     
    int currentValue = 3; 
     std::function<int (int)> moveNearerToZero = chooseStepFunction(currentValue > 0);
 }